WcaOnRails/lib/custom_i18n_scanner.rb

```diff
@@ -0,0 +1,82 @@
+require 'i18n/tasks/scanners/file_scanner'
+class WCAFileScanner < I18n::Tasks::Scanners::FileScanner
+  include I18n::Tasks::Scanners::RelativeKeys
+  include I18n::Tasks::Scanners::Results
+
+  def active_record?(model)
+    # FIXME: Find a way to determine this dynamically ?
+    model != "contact"
+  end
+
+  def extract_model_attr(key)
+    model_key = key.gsub(/^(\w+)\..*$/, '\1').singularize
+    # Some dirty tricks to assign the detected field to the "correct" model
+    # For most controllers the forms modify a model which has the same name
+    # as the controller, however for some of them they modified a single different
+    # model (eg: the `admin` forms modify `person`s, and only that model).
+    model_name = case model_key
+                 when "devise" then "user"
+                 when "admin" then "person"
+                 when "oauth" then "doorkeeper/application"
+                 else model_key
+                 end
+    attribute = key.gsub(/.*\.(\w+)$/, '\1')
+    [model_name, attribute]
+  end
+
+  def occurrence_from_line(path, line, line_number, position)
+    Occurrence.new(
+      path:     path,
+      pos:      line_number + position,
+      line_num: line_number,
+      line_pos: position,
+      line:     line,
+      raw_key: nil)
+  end
+
+  # @return [Array<[absolute key, Results::Occurrence]>]
+  def scan_file(path)
+    text = read_file(path)
+    retval = []
+    line_number = 0
+    text.each_line do |line|
+      line_number += 1
+      # Scan for possible radio choices
+      if line =~ /^\s*<%= f.input :(\w+), as: :radio_buttons, collection: \[(.*)\](.*)%>/
+        match = Regexp.last_match
+        abskey = absolute_key(".#{match[1]}", path)
+        occurrence = occurrence_from_line(path, line, line_number, match.offset(0).first)
+        model, attribute = extract_model_attr(abskey)
+        # Get all choices from the f.input
+        options = match[2].gsub(/[: ]/, '').split(',')
+        options.each do |o|
+          # Mark every choice as used
+          retval << ["simple_form.options.#{model}.#{attribute}.#{o}", occurrence]
+        end
+      end
+      # We don't use an else here as we want to match radios here too!
+      # Scan for inputs (or labels for inputs) that are not hidden
+      next unless line =~ /^\s*<%= f.(input|label|input_field) :(\w+)(?!.*hidden)(.*)%>/
```

Comment:
I have a ruby question about this one; Rubocop complained that I was doing something like this:
```ruby
loop
  if a
    #do something
  end
  if b
    #do something else
  end
end
```

and wanted me to rewrite it like this:
```ruby
loop
  if a
    #do something
  end
next unless b
#do something else
end
```

Is there any trick with Ruby's implementation justifying this?
Because I don't find it easier to read...